builder/remotecontext/detect.go

```diff
@@ -129,11 +129,11 @@ func removeDockerfile(c modifiableContext, filesToRemove ...string) error {
 	case err != nil:
 		return err
 	}
+	defer f.Close()
```

Comment:
Not sure this will work; f will no longer be closed before it's being removed in the `filesToRemove` loop below

Perhaps add `f.Close()` inside the `if err != nil {` instead?